### PR TITLE
Don't offer to generate parameters in lambdas

### PR DESCRIPTION
**Pull request by davidwengier**
Fixes https://github.com/dotnet/roslyn/issues/47586

### CODE REVIEW

File: src/Features/Core/Portable/GenerateMember/GenerateVariable/AbstractGenerateVariableService.State.cs [hunk 1]
```diff
@@ -151,7 +151,7 @@ internal bool CanGenerateParameter()
                 // !this.IsInMemberContext prevents us offering this fix for `x.goo` where `goo` does not exist
                 // Workaround: The compiler returns IsImplicitlyDeclared = false for <Main>$.
                 return ContainingMethod is { IsImplicitlyDeclared: false, Name: not WellKnownMemberNames.TopLevelStatementsEntryPointMethodName }
-                    && !IsInMemberContext && !IsConstant;
+                    && !IsInMemberContext && !IsConstant && !ContainingMethod.IsAnonymousFunction();
```
Comment (CyrusNajmabadi) [inline src/Features/Core/Portable/GenerateMember/GenerateVariable/AbstractGenerateVariableService.State.cs:154]: not exactly what i woudl expect.  i woudl still think we'd offer to add the parameter, but just inside AssertSomething.

Comment (davidwengier) [reply src/Features/Core/Portable/GenerateMember/GenerateVariable/AbstractGenerateVariableService.State.cs:154]: By "inside" do you mean add the parameter to the lambda, or add the parameter to the AssertSomething method?

Comment (davidwengier) [reply src/Features/Core/Portable/GenerateMember/GenerateVariable/AbstractGenerateVariableService.State.cs]: Took a punt you meant the latter

Comment (CyrusNajmabadi) [reply src/Features/Core/Portable/GenerateMember/GenerateVariable/AbstractGenerateVariableService.State.cs]: yup, the latter.

**Review by ryzngard — APPROVED** [review]

**Pull request merged**